File: CoreAngularDemo.Client/src/app/modules/shared/models/part-in.ts
import { TEntity } from '../../core/models/entity/entity';
import { Currency } from './currency';
import { Part } from './part';
import { Unit } from './unit';

export class PartIn extends TEntity<PartIn> {
  createdDate: Date;
  updatedDate: Date;
  arrivalDate: string;
  amount: number;
  price: number;
  batch: string;

  currency: Currency;
  part: Part;
  unit: Unit;

  partName: string;
  unitName: string;
  currencyName: string;

  constructor(partIn: Partial<PartIn>) {
    super(partIn);
    this.part = partIn.part;
    this.unit = partIn.unit;
    this.currency = new Currency(partIn.currency);
    if (this.currency) {
      this.currencyName = this.currency.fullName;
    }
    if (this.part) {
      this.partName = partIn.part.name;
    }
    if (this.unit) {
      this.unitName = partIn.unit.name;
    }
  }
}

File: CoreAngularDemo.API/CoreAngularDemo.DAL/Models/Entities/PartIn.cs
using System;
using CoreAngularDemo.DAL.Models.Entities.Abstractions;

namespace CoreAngularDemo.DAL.Models.Entities
{
    /// <summary>
    /// Parts deliveries (поставки деталей).
    /// </summary>
    public class PartIn : IAuditableEntity, IBaseEntity
    {
        public PartIn()
        {

        }

        public int Id { get; set; }
        public uint Amount { get; set; }
        public float Price { get; set; }
        public DateTime ArrivalDate { get; set; }

        /// <summary>
        /// Номер партії.
        /// </summary>
        public string Batch { get; set; }
        public int UnitId { get; set; }
        public int PartId { get; set; }
        public int CurrencyId { get; set; }
        public string CreatedById { get; set; }
        public string UpdatedById { get; set; }
        public DateTime? CreatedDate { get; set; }
        public DateTime? UpdatedDate { get; set; }

        public Currency Currency { get; set; }
        public Unit Unit { get; set; }
        public Part Part { get; set; }
    }
}
